editor/animation_track_editor.cpp

```diff
@@ -3790,8 +3790,9 @@ void AnimationTrackEditor::insert_value_key(const String &p_property, const Vari
 				value = p_value; //all good
 			} else {
 				String tpath = animation->track_get_path(i);
-				if (NodePath(tpath.get_basename()) == np) {
-					String subindex = tpath.get_extension();
+				int index = tpath.find_last(":");
+				if (NodePath(tpath.substr(0, index + 1)) == np) {
+					String subindex = tpath.substr(index + 1, tpath.length() - index);
```

Comment:
You can also use `left()` and `right()` which are somewhat more convenient for this.
Arewe sure that there will always be a `:`?

Reply:
yes. it comes from here
https://github.com/godotengine/godot/blob/3bcedd02b2eb2cfb78327249cafa195b55cdc499/editor/animation_track_editor.cpp#L3907-L3954